src/modules/auth/component/SignUp/SignUp.tsx

```diff
@@ -74,54 +78,81 @@ export function SignUp() {
+                        type="email"
+                        {...field}
+                      />
+                    )}
                   />
+
                 </Grid>
                 <Grid item xs={12}>
-                  <TextField
-                    label="Seller name"
-                    placeholder="Seller name"
-                    type="text"
-                    size="small"
-                    fullWidth
+                  <Controller
                     name="name"
-                    value={formData.name}
-                    onChange={handleChange}
+                    control={control}
+                    rules={{ required: "Name is required" }}
+                    render={({ field }) => (
+                      <TextField
+                        label="Seller name"
+                        placeholder="Seller name"
+                        type="text"
+                        {...field}
+                      />
+                    )}
                   />
                 </Grid>
                 <Grid item xs={12}>
-                  <TextField
-                    label="Password"
-                    placeholder="********"
-                    type="password"
-                    size="small"
-                    fullWidth
+                  <Controller
                     name="password"
-                    value={formData.password}
-                    onChange={handleChange}
+                    control={control}
+                    rules={{ required: "Password is required", minLength: { value: 8, message: "Password must be at least 8 characters long" } }}
+                    render={({ field }) => (
+                      <TextField
+                        label="Password"
+                        placeholder="********"
+                        type="password"
+                        {...field}
+                      />
+                    )}
                   />
                 </Grid>
                 <Grid item xs={12}>
-                  <TextField
-                    label="Confirm password"
-                    placeholder="********"
-                    type="password"
-                    size="small"
-                    fullWidth
+                  <Controller
                     name="confirmPassword"
-                    value={formData.confirmPassword}
-                    onChange={handleChange}
+                    control={control}
+                    rules={{ required: "Please confirm your password", validate: (value) => value === getValues("password") || "Passwords do not match" }}
+                    render={({ field }) => (
+                      <TextField
+                        label="Confirm password"
+                        placeholder="********"
+                        type="password"
+                        {...field}
+                      />
+                    )}
                   />
                 </Grid>
+                {errors.email && (
```

Comment:
TextField has helperText and error props. You need to use them both
Here is the [helperText](https://mui.com/material-ui/api/text-field/#text-field-prop-helperText), that should just display the text under the TextField

Here is the [error](https://mui.com/material-ui/api/text-field/#text-field-prop-error), that is just a boolean. You can pass there the Boolean(errors.name) for example and it will indicate that the helperText should be red

Reply:
done